rfc/rfc-0000-add-template-field-to-runnable.md

```diff
@@ -0,0 +1,195 @@
+
+```yaml
+---
+apiVersion: carto.run/v1alpha1
+kind: ClusterRunTemplate
+metadata:
+  name: md-linting-pipelinerun
+spec:
+  template:
+    apiVersion: tekton.dev/v1beta1
+    kind: PipelineRun
+    metadata:
+      generateName: $(runnable.metadata.name)$-pipeline-run-
+    spec:
+      pipelineRef:
+        name: linter-pipeline
+      params:
+        - name: repository
+          value: $(runnable.spec.inputs.repository)$
+        - name: revision
+          value: $(runnable.spec.inputs.revision)$
+      workspaces:
+        - name: shared-workspace
+          volumeClaimTemplate:
+            spec:
+              accessModes:
+                - ReadWriteOnce
+              resources:
+                requests:
+                  storage: 256Mi
+  outputs:
+    url: spec.params[?(@.name=="repository")].value
+    revision: spec.params[?(@.name=="revision")].value
+
+---
+apiVersion: carto.run/v1alpha1
+kind: Runnable
+metadata:
+  name: linter
+spec:
+  runTemplateRef:
+    name: md-linting-pipelinerun
+  inputs:
+    repository: some-repo
+    revision: some-sha
+  serviceAccountName: pipeline-run-management-sa
+```
+
+This would become the single Runnable here:
+
+```yaml
+apiVersion: carto.run/v1alpha1
+kind: Runnable
+metadata:
+  name: linter
+spec:
+  serviceAccountName: pipeline-run-management-sa
+  template:
+    apiVersion: tekton.dev/v1beta1
+    kind: PipelineRun
+    metadata:
+      generateName: some-name-pipeline-run-
+    spec:
+      pipelineRef:
+        name: linter-pipeline
+      params:
+        - name: repository
+          value: some-repo
+        - name: revision
+          value: some-sha
+      workspaces:
+        - name: shared-workspace
+          volumeClaimTemplate:
+            spec:
+              accessModes:
+                - ReadWriteOnce
+              resources:
+                requests:
+                  storage: 256Mi
+  outputs:
+    url: spec.params[?(@.name=="repository")].value
+    revision: spec.params[?(@.name=="revision")].value
+```
+
+The target persona is Cartographer maintainers; the main driver of this is to enable app operators to create templates
+that leverage Runnable without needing to submit/manage ClusterRunTemplates.
+See [this RFC](https://github.com/vmware-tanzu/cartographer/pull/705) for discussion of that use case.
+
+# How it Works
+
+[how-it-works]: #how-it-works
+
+The object defined in the Runnable is submitted to the cluster. The runnable exposes the output fields defined in
+the runnable's own status. Care should be taken to garbage collect objects submitted by a runnable if the GVK or
+name of the object is changed.
+
+# Migration
+
+[migration]: #migration
+
+See [spec-changes].
+
+This RFC would require a version bump for Runnable. There is no technical reason that Runnables with valid
+references to ClusterRunTemplate could not be translated into Runnables with template fields embedded.
```

Comment:
Bumping the version isn't required for additive changes. If we did update the version, the old and new fields would need to have a home on both the v1alpha1 and v1alpha2 versions, in order for the CRD conversion support to function.